### ISSUE TITLE
links not working on specific Android builds

### ISSUE THREAD

**Issue by WesselvanDam** [flutter/flutter#148090]
### What package does this bug report belong to?

url_launcher

### What target platforms are you seeing this bug on?

Android

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
  url_launcher:
    dependency: "direct main"
    description:
      name: url_launcher
      sha256: "6ce1e04375be4eed30548f10a315826fd933c1e493206eab82eed01f438c8d2e"
      url: "https://pub.dev"
    source: hosted
    version: "6.2.6"
  url_launcher_android:
    dependency: transitive
    description:
      name: url_launcher_android
      sha256: "31222ffb0063171b526d3e569079cf1f8b294075ba323443fdc690842bfd4def"
      url: "https://pub.dev"
    source: hosted
    version: "6.2.0"
```

</details>

### Steps to reproduce

When building a Flutter app on Android (either debug or release) with url_launcher links, the links do not open. They used to work before, but either an update in Flutter, the url_launcher package itself or in Flutter SDK 34 (which I recently migrated to) caused the issue. However, if I have team members build the same app with their devices on my phone, the links do work. It is therefore difficult to reproduce the error, but I use the same version of Flutter, url_launcher, etc. as they do so I do not understand what the problem is here.

### Expected results

Link is opened in the browser

### Actual results

Link is not opened. Instead, an error message appears in the logs (see below)

### Code sample

N/A

### Screenshots or Videos

N/A

### Logs

<details open><summary>Logs</summary>

```console
E/flutter (19111): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(channel-error, Unable to establish connection on channel., null, null)
E/flutter (19111): #0      UrlLauncherApi.launchUrl (package:url_launcher_android/src/messages.g.dart:118:7)
E/flutter (19111): <asynchronous suspension>
E/flutter (19111): #1      UrlLauncherAndroid.launchUrl (package:url_launcher_android/url_launcher_android.dart:113:11)
E/flutter (19111): <asynchronous suspension>
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.19.6, on Microsoft Windows [Version 10.0.22631.3447], locale en-GB)
    • Flutter version 3.19.6 on channel stable at C:\Users\wesse\fvm\versions\3.19.6
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 54e66469a9 (3 weeks ago), 2024-04-17 13:08:03 -0700
    • Engine revision c4cd48e186
    • Dart version 3.3.4
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\wesse\AppData\Local\Android\Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = C:\Users\wesse\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[!] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.1)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.8.34316.72
    ! Unable to locate a Windows 10 SDK. If building fails, install the Windows 10 SDK in Visual Studio.

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code (version 1.90.0-insider)
    • VS Code at C:\Users\wesse\AppData\Local\Programs\Microsoft VS Code Insiders
    • Flutter extension version 3.88.0

[√] Connected device (4 available)
    • SM A715F (mobile) • R58N51AWRTP • android-arm64  • Android 13 (API 33)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [Version 10.0.22631.3447]
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 124.0.6367.156
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 124.0.2478.80

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by huycozy** [comment]
Hi @WesselvanDam 
I checked this issue on my Android 14 device using [package example](https://github.com/flutter/packages/tree/main/packages/url_launcher/url_launcher/example) but couldn't reproduce the error. Could you try with this example and confirm?

Also, I see the fetched `url_launcher_android` is not up-to-date. You can try upgrading to the latest version `url_launcher_android: ^6.3.1` and retry.

**Comment by WesselvanDam** [comment]
Thanks for the reply. I tried the package example, and url's work as expected there; with the following pubspec.lock:

<details><summary>pubspec.lock example app</summary>
  url_launcher:
    dependency: "direct main"
    description:
      path: ".."
      relative: true
    source: path
    version: "6.2.6"
  url_launcher_android:
    dependency: transitive
    description:
      name: url_launcher_android
      sha256: "360a6ed2027f18b73c8d98e159dda67a61b7f2e0f6ec26e86c3ada33b0621775"
      url: "https://pub.dev"
    source: hosted
    version: "6.3.1"
</details>

In my app, when changing the url_launcher pubspec version to 6.2.6, it does not bump the _android version to 6.3.1 but to 6.2.0. Making it a direct dependency (instead of transitive) and setting it to 6.3.1 does not fix the issue on my app. 

So in short:

1.  Using my laptop and my Android phone, my app doesn't work
2.  Using my laptop and my Android phone, the example app does work
3.  Using a different laptop and my Android phone, my app does work

Which is a strange set of observations. 
Android version is 13 btw.

**Comment by huycozy** [comment]
Do you use `pigeon` package? Could you share dependencies on your project? (pubspec.yaml, pubspec.lock) (It may too long so please put it inside the collapse section as you did above, thanks)

**Comment by WesselvanDam** [comment]
I do not use `pigeon`. Here's my dependencies:

<details><summary>pubspec.yaml</summary>
dependencies:
  animations: ^2.0.2
  async: ^2.11.0
  blurhash_dart: ^1.2.1
  cached_network_image: ^3.2.1
  cloud_firestore: ^4.15.5
  cloud_functions: ^4.6.0
  connectivity_plus: ^2.0.3
  confetti: ^0.7.0
  crop_image: ^1.0.11
  cupertino_icons: ^1.0.3
  envied: ^0.5.2
  firebase_auth: ^4.15.5
  firebase_core: ^2.25.4
  firebase_messaging: ^14.0.1
  firebase_remote_config: ^4.2.3
  firebase_storage: ^11.5.6
  firebase_ui_firestore: ^1.6.2
  flutter:
    sdk: flutter
  flutter_blurhash: ^0.8.2
  flutter_local_notifications: ^16.1.0
  flutter_localizations:
    sdk: flutter
  flutter_quill: 9.3.4
  flutter_quill_extensions: 9.3.4
  flutter_riverpod: ^2.5.1
  freezed_annotation: ^2.2.0
  go_router: ^13.2.4
  google_fonts: ^4.0.3
  googleapis: ^11.2.0
  googleapis_auth: ^1.4.1
  http: ^1.0.0
  image: ^4.1.7
  image_cropper: ^5.0.0
  image_picker: ^1.0.4
  intl: ^0.18.0
  keyboard_dismisser: ^3.0.0
  package_info_plus: ^4.0.2
  path_provider: ^2.0.1
  riverpod_annotation: ^2.3.5
  share_plus: ^7.0.2
  shared_preferences: ^2.0.0
  slang: ^2.7.0
  slang_flutter: ^2.7.0
  stack_trace: ^1.11.0
  table_calendar: ^3.0.0
  timezone: ^0.9.1
  tuple: ^2.0.1
  url_launcher: ^6.0.6
  url_strategy: ^0.2.0
  visibility_detector: ^0.4.0+2

dependency_overrides:
  super_native_extensions: ^0.8.10
  device_info_plus: ^10.0.1


dev_dependencies:
  flutter_test:
    sdk: flutter
  lint: ^2.1.2
  slang_build_runner: ^2.7.0
  build_runner: ^2.2.0
  fake_cloud_firestore: ^2.4.9
  freezed: ^2.4.7
  mockito: ^5.4.4
  riverpod_generator: ^2.3.5
  custom_lint: ^0.5.4
  riverpod_lint: ^2.3.2
  envied_generator: ^0.5.2
</details>

The .lock as a text file because it is quite large indeed:
[pubspec.txt](https://github.com/flutter/flutter/files/15274685/pubspec.txt)

**Comment by huycozy** [comment]
<details open>
<summary>url_launcher in pubspec.lock</summary>

```lock
url_launcher:
  dependency: "direct main"
  description:
    name: url_launcher
    sha256: "0ecc004c62fd3ed36a2ffcbe0dd9700aee63bd7532d0b642a488b1ec310f492e"
    url: "https://pub.dev"
  source: hosted
  version: "6.2.5"
url_launcher_android:
  dependency: transitive
  description:
    name: url_launcher_android
    sha256: "31222ffb0063171b526d3e569079cf1f8b294075ba323443fdc690842bfd4def"
    url: "https://pub.dev"
  source: hosted
  version: "6.2.0"
```
</details>

I see `url_launcher_android` is not up-to-date. Could you try upgrading it to the latest version `url_launcher_android: ^6.3.1` and retry? If the issue persists, please share the entire output of `flutter run -v`, thanks!

**Comment by WesselvanDam** [comment]
Apologies, in a orevious comment I stated:

> In my app, when changing the url_launcher pubspec version to 6.2.6, it does not bump the _android version to 6.3.1 but to 6.2.0. Making it a direct dependency (instead of transitive) and setting it to 6.3.1 does not fix the issue on my app.

In the pubspec.yaml and lock I attached later, the versions were reset and showed the older dependencies. Using url_launcher_android 6.3.1 does not fix the problem.

I'll see if I can share the output of flutter run -v later today!

**Comment by WesselvanDam** [comment]
@huycozy here it is:

[log.txt](https://github.com/flutter/flutter/files/15286979/log.txt)

Not seeing anything strange right away though

**Comment by darshankawar** [comment]
@WesselvanDam 
I looked at the dependencies you are using and the issue could be occuring due to third party package conflict that you'll need to verify by narrowing down the issue and by removing the packages one by one and see which one could be leading to the error. I suspect `flutter_quill` and `shared_preferences` could be the one.  See [this](https://github.com/flutter/flutter/issues/147824#issuecomment-2096038631) for reference and check if it helps in your case or not.

**Comment by WesselvanDam** [comment]
@darshankawar thank you for the leads! The linked issue led me to uninstall Rust (which I didn't need anyway) from my device, and that alone fixed the issue already. It does seem that an updated version of Rust may have solved the issue as well, but as I didn't need it anymore I decided to just uninstall it. Thanks again!
